tests/performance/reindexing/reindexing_and_feeding.rb

```diff
@@ -0,0 +1,254 @@
+      result.add_metric('reindexing.throughput', document_count * 1e3 / time_millis)
+      result.add_parameter('legend', concurrent_operations)
+    end
+  end
+
+  # Feed data with the given config, which must include :file.
+  def feed_data(config)
+    profiler_start if config.has_key?(:legend)
+    run_feeder(config[:file], [], config.merge({ :localfile => true, :numthreads => 8 }))
+    profiler_report(config[:legend]) if config.has_key?(:legend)
+  end
+
+  # Wait for reindexing after the given time to have started.
+  def wait_for_reindexing_start(ready_millis)
+    while true
+      status = get_reindexing_status
+      break if status and status['startedMillis'] > ready_millis
+      sleep 1
+    end
+  end
+
+  # Wait for reindexing to successfully complete, and return the time used in milliseconds.
+  def wait_for_reindexing
+    while true
+      status = get_reindexing_status
+      break if status and ["successful", "failed"].include? status['state']
+      sleep 1
+    end
+    assert("successful" == status['state'], "Reindexing should complete successfully")
+    return status['endedMillis'] - status['startedMillis']
+  end
+
+  # Fetch reindexing status from reindexing controller.
+  def get_reindexing_status
+    status = vespa.clustercontrollers["0"].get_reindexing_json
+    return nil if status.nil?
+    return status['status'].first
+  end
+
+
+  def get_graphs
+    @test_config.map do |config, metrics|
+      metrics.map do |metric, limits|
+        {
+          :x => "blank",
+          :y => metric,
+          :title => " #{config[:legend]} #{metric}",
+          :filter => { "legend" => config[:legend] },
+          :historic => true
+        }.merge(limits)
+      end
+    end.flatten
+  end
+
+  # Wait for convergence of all services in the application — specifically document processors 
+  def wait_for_convergence(generation)
+    start_time = Time.now
+    until get_json(http_request(URI(application_url + "serviceconverge"), {}))["converged"] or Time.now - start_time > 60 # seconds
+      sleep 1
+    end
+    assert(Time.now - start_time < 60, "Services should converge on new generation within the minute")
+    assert(generation == get_json(http_request(URI(application_url + "serviceconverge"), {}))["wantedGeneration"],
+           "Should converge on generation #{generation}")
+    puts "Services converged on new config generation after #{Time.now - start_time} seconds"
+  end
+
+  # Trigger reindexing of the whole corpus
+  def trigger_reindexing
+    # Read baseline reindexing status — very first reindexing is a no-op in the reindexer controller
+    response = http_request(URI(application_url + "reindexing"), {})
+    assert(response.code.to_i == 200, "Request should be successful")
+    previous_reindexing_timestamp = get_json(response)["status"]["readyMillis"]
+
+    # Allow the reindexing maintainer some time to run, and mark the first no-op reindexing as done
+    sleep 60
```

Comment:
Can this be replaced by periodic polling of some status API?

Reply:
Oops, leftover from old code.